langchain/evaluation/run_evaluators/base.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Protocol
+
+from langchainplus_sdk.evaluation.evaluator import EvaluationResult
+from langchainplus_sdk.schemas import Example, Run
+
+from langchain.callbacks.manager import CallbackManagerForChainRun
+from langchain.chains.base import Chain
+from langchain.chains.llm import LLMChain
+from langchain.schema import BaseOutputParser
+
+
+class RunEvalInputMapper(Protocol):
+    """Map the inputs of a run to the inputs of an evaluation."""
+
+    def map(self, run: Run, example: Optional[Example] = None) -> Dict[str, Any]:
+        """Maps the Run and Optional[Example] to a dictionary"""
+
+
+class RunEvaluatorOutputParser(BaseOutputParser[EvaluationResult]):
+    """Parse the output of a run."""
+
+    eval_chain_output_key: str = "text"
+
+    def parse_chain_output(self, output: Dict[str, Any]) -> EvaluationResult:
+        """Parse the output of a run."""
+        text = output[self.eval_chain_output_key]
+        return self.parse(text)
+
+
+class RunEvaluator(Chain):
```

Comment:
won't this clash with RunEvaluator from the sdk? Also most (all?) subclasses of `Chain` end with `Chain`, i.e. `RunEvaluatorChain`

Reply:
Actually I'm a bit confused here. Should this just be a subclass of `langchainplus_sdk.evaluation.evaluator.RunEvaluator`? It implements `evaluate_run`

Reply:
It should! my oversight - will rename and inherit